src/SonarScanner.MSBuild.PreProcessor/PreProcessor.cs

```diff
@@ -31,23 +31,23 @@ public sealed class PreProcessor : IPreProcessor
     private static readonly string[] Languages = [CSharpLanguage, VBNetLanguage];
 
     private readonly IPreprocessorObjectFactory factory;
-    private readonly ILogger logger;
+    private readonly IRuntime runtime;
 
-    public PreProcessor(IPreprocessorObjectFactory factory, ILogger logger)
+    public PreProcessor(IPreprocessorObjectFactory factory, IRuntime runtime)
     {
         this.factory = factory ?? throw new ArgumentNullException(nameof(factory));
-        this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        this.runtime = runtime ?? throw new ArgumentNullException(nameof(runtime));
     }
 
     public async Task<bool> Execute(IEnumerable<string> args)
     {
-        logger.SuspendOutput();
-        var processedArgs = ArgumentProcessor.TryProcessArgs(args, logger);
+        runtime.Logger.SuspendOutput();
```

Comment:
Just because you're touching this line, please add a comment.
I've added the same to the PostProcessor. It took me way too long to figure out why it is actually needed
```suggestion
        runtime.Logger.SuspendOutput(); // Wait for the correct verbosity to be calculated
```